src/types/closure.md

```diff
@@ -178,3 +288,117 @@ Because captures are often by reference, the following general rules arise:
+
+```rust
+{
+    let tuple = 
+      (String::from("foo"), String::from("bar")); // --+
+    { //                                               |
+        let c = || { // ----------------------------+  |
+            // tuple.0 is captured into the closure |  |
+            drop(tuple.0); //                 |  |
+        }; //                                       |  |
+    } // 'c' and 'tuple.0' dropped here ------------+  |
+} // tuple.1 dropped here -----------------------------+
+```
+
+## Wild Card Patterns
+Closures now only capture data that needs to be read, which means the following closures will not capture `x`
+
+```rust
+let x = 10;
+let c = || {
+    let _ = x;
+};
+
+let c = || match x {
+    _ => println!("Hello World!")
+};
+```
+
+
+# Edition 2018 and before
+
+## Closure types difference
+
+In Edition 2018 and before, a closure would capture variables in its entirety. This meant that for the example used in the [Closure types]() section the generated closure type would look something like this:
+
+<!-- ignore: simplified -->
+```rust,ignore
+struct Closure<'a> {
+    rect : &'a mut Rectangle,
+}
+
+impl<'a> FnOnce<()> for Closure<'a> {
+    type Output = String;
+    fn call_once(self) -> String {        
+        self.rect.left_top.x += 1;
+        self.rect.right_bottom.x += 1;
+        format!("{:?}", self.rect.left_top)
+    }
+}
+```
+and the call to `f` would work as if it were:
+<!-- ignore: continuation of above -->
+```rust,ignore
+f(Closure { rect: rect });
+```
+
+## Capture precision difference
+
+Composite types such as structs, tuples, and enums are always captured in its intirety,
+not by individual fields. It may be necessary to borrow into a local variable in order to capture a single field:
+
+```rust
+# use std::collections::HashSet;
+#
+struct SetVec {
+    set: HashSet<u32>,
+    vec: Vec<u32>
+}
+
+impl SetVec {
+    fn populate(&mut self) {
+        let vec = &mut self.vec;
+        self.set.iter().for_each(|&n| {
+            vec.push(n);
+        })
+    }
+}
+```
+
+If, instead, the closure were to use `self.vec` directly, then it would attempt
+to capture `self` by mutable reference. But since `self.set` is already
+borrowed to iterate over, the code would not compile.
+
+If the `move` keyword is used, then all captures are by move or, for `Copy`
+types, by copy, regardless of whether a borrow would work. The `move` keyword is
+usually used to allow the closure to outlive the captured values, such as if the
+closure is being returned or used to spawn a new thread.
+
+Regardless of if the data will be read by the closure, i.e. in case of wild card patterns, if a variable defined outside the closure is mentioned within the closure the variable will be captured in its entirety.
+
+## Other traits difference
```

Comment:
I think we can actually delete this section. This is because the modified rules stated "the captured values" and we already clarified in the earlier Edition2018 subsection that the values captured are variables that are captured in its entirety

Reply:
I'm fine removing it